streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/ChangeEventBus.scala

```diff
@@ -24,11 +24,11 @@ import java.util.concurrent.{LinkedBlockingQueue, ThreadPoolExecutor, TimeUnit}
 class ChangeEventBus {
 
   private val execPool = new ThreadPoolExecutor(
-    Runtime.getRuntime.availableProcessors * 10,
-    Runtime.getRuntime.availableProcessors * 20,
+    Runtime.getRuntime.availableProcessors * 2,
```

Comment:
A small question for me, why is the CPU and buffer queue configuration of the entire ChangeEventBus thread pool narrowed down here?

Reply:
I think the efficiency has been greatly improved by turning it into multi-threaded asynchronous, so there is no need for such high configuration.

Reply:
I think maybe we need a proof that it actually efficient when deploy more jobs. What do you think about it? @Al-assad

Reply:
Thank you for your replies. I will try to conduct a detailed test and give the test results. If you are convenient, you can also do some tests, so that the results can be used for some comparisons to make the results more accurate.

Reply:
> most of the subscription programs in this part are blocking IO

Hi @Al-assad  I don't quite understand this sentence..  I don't know if your understanding of blocking is consistent with my understanding. My understanding is that after adding `@AllowConcurrentEvents` this annotation will create `Subscriber` instead of `SynchronizedSubscriber`.

In `SynchronizedSubscriber` it is because of the use of `synchronize` that the blocking effect is achieved.
```
@Override
  void invokeSubscriberMethod(Object event) throws InvocationTargetException {
    synchronized (this) {
      super.invokeSubscriberMethod(event);
    }
  }
```

Reply:
> A small question for me, why is the CPU and buffer queue configuration of the entire ChangeEventBus thread pool narrowed down here?

Hi @Al-assad  . Let me add my understanding on this point. `Runtime.getRuntime().availableProcessors()` gets the number of CPU core threads.  If the number of enabled threads is much greater than the number of CPU core threads, the cost of context switching will be increased, which will affect performance instead.

Not sure if my understanding is correct. looking forward to your reply  :）

Reply:
@Al-assad  Thanks for the information you provided, I have modified the thread pool parameters. There is also a discussion about blocking IO here.:eyes: